## Задача 1.

1. Скачать файл в уроке
2. Считать данные с помощью pandas
3. Вывести на экран первые 5 строк
4. Посмотреть на описание признаков и на их содержание

In [2]:
import pandas as pd
import numpy as np
house = pd.read_csv('kc_house_data.csv')
house

id             date     price  bedrooms  bathrooms  \
0      7129300520  20141013T000000  221900.0         3       1.00   
1      6414100192  20141209T000000  538000.0         3       2.25   
2      5631500400  20150225T000000  180000.0         2       1.00   
3      2487200875  20141209T000000  604000.0         4       3.00   
4      1954400510  20150218T000000  510000.0         3       2.00   
...           ...              ...       ...       ...        ...   
21608   263000018  20140521T000000  360000.0         3       2.50   
21609  6600060120  20150223T000000  400000.0         4       2.50   
21610  1523300141  20140623T000000  402101.0         2       0.75   
21611   291310100  20150116T000000  400000.0         3       2.50   
21612  1523300157  20141015T000000  325000.0         2       0.75   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
0             1180      5650     1.0           0     0  ...      7   
1             2570      7242     2.0           0     0  ...      7   
2              770     10000     1.0           0     0  ...      6   
3             1960      5000     1.0           0     0  ...      7   
4             1680      8080     1.0           0     0  ...      8   
...            ...       ...     ...         ...   ...  ...    ...   
21608         1530      1131     3.0           0     0  ...      8   
21609         2310      5813     2.0           0     0  ...      8   
21610         1020      1350     2.0           0     0  ...      7   
21611         1600      2388     2.0           0     0  ...      8   
21612         1020      1076     2.0           0     0  ...      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
0            1180              0      1955             0    98178  47.5112   
1            2170            400      1951          1991    98125  47.7210   
2             770              0      1933             0    98028  47.7379   
3            1050            910      1965             0    98136  47.5208   
4            1680              0      1987             0    98074  47.6168   
...           ...            ...       ...           ...      ...      ...   
21608        1530              0      2009             0    98103  47.6993   
21609        2310              0      2014             0    98146  47.5107   
21610        1020              0      2009             0    98144  47.5944   
21611        1600              0      2004             0    98027  47.5345   
21612        1020              0      2008             0    98144  47.5941   

          long  sqft_living15  sqft_lot15  
0     -122.257           1340        5650  
1     -122.319           1690        7639  
2     -122.233           2720        8062  
3     -122.393           1360        5000  
4     -122.045           1800        7503  
...        ...            ...         ...  
21608 -122.346           1530        1509  
21609 -122.362           1830        7200  
21610 -122.299           1020        2007  
21611 -122.069           1410        1287  
21612 -122.299           1020        1357  

[21613 rows x 21 columns]

In [15]:
house.head()

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  

[5 rows x 21 columns]

- id - Уникальный ID для каждого дома
- date - Дата продажи дома
- price - Стоимость продажи дома
- bedrooms - Кол-во спален
- bathrooms - Кол-во ванных комнат (0.5 - туалет без душа)
- sqft_living - Кв. метры жилые
- sqft_lot - Кв. метры общие
- floors - Кол-во этажей
- waterfront - Есть набержная или нет
- view - Значение от 0 до 4 насколько хороший вид
- condition - Значение от 1 до 5 насколько хорошее состояние
- grade - Значение от 1 до 13, где 1-3 плохая конструкция здания и дизайн, 7 - средний уровень конструкции и дизайна, 11-13 - высокое качество конструкции и дизайна
- sqft_above - Кв. метры дома, которые находятся выше земли
- sqft_basement - Кв. метры дома, которые находятся ниже земли
- yr_built - Год постройки дома
- yr_renovated - Год ремонта дома
- zipcode - Индекс
- lat - Широта
- long - Долгота
- sqft_living15 - Кв. метры жилой площади у 15 соседей
- sqft_lot15 - Кв. метры общей площади у 15 соседей

## Задача 2.

1. Проведите первичный анализ данных
2. Изучите типы данных
3. Найдите количество пропущенных ячеек в данных
4. Посчитайте основные статистики по всем признакам и поизучайте их
5. Пишите выводы

In [16]:
house.describe()

id         price      bedrooms     bathrooms   sqft_living  \
count  2.161300e+04  2.161300e+04  21613.000000  21613.000000  21613.000000   
mean   4.580302e+09  5.400881e+05      3.370842      2.114757   2079.899736   
std    2.876566e+09  3.671272e+05      0.930062      0.770163    918.440897   
min    1.000102e+06  7.500000e+04      0.000000      0.000000    290.000000   
25%    2.123049e+09  3.219500e+05      3.000000      1.750000   1427.000000   
50%    3.904930e+09  4.500000e+05      3.000000      2.250000   1910.000000   
75%    7.308900e+09  6.450000e+05      4.000000      2.500000   2550.000000   
max    9.900000e+09  7.700000e+06     33.000000      8.000000  13540.000000   

           sqft_lot        floors    waterfront          view     condition  \
count  2.161300e+04  21613.000000  21613.000000  21613.000000  21613.000000   
mean   1.510697e+04      1.494309      0.007542      0.234303      3.409430   
std    4.142051e+04      0.539989      0.086517      0.766318      0.650743   
min    5.200000e+02      1.000000      0.000000      0.000000      1.000000   
25%    5.040000e+03      1.000000      0.000000      0.000000      3.000000   
50%    7.618000e+03      1.500000      0.000000      0.000000      3.000000   
75%    1.068800e+04      2.000000      0.000000      0.000000      4.000000   
max    1.651359e+06      3.500000      1.000000      4.000000      5.000000   

              grade    sqft_above  sqft_basement      yr_built  yr_renovated  \
count  21613.000000  21613.000000   21613.000000  21613.000000  21613.000000   
mean       7.656873   1788.390691     291.509045   1971.005136     84.402258   
std        1.175459    828.090978     442.575043     29.373411    401.679240   
min        1.000000    290.000000       0.000000   1900.000000      0.000000   
25%        7.000000   1190.000000       0.000000   1951.000000      0.000000   
50%        7.000000   1560.000000       0.000000   1975.000000      0.000000   
75%        8.000000   2210.000000     560.000000   1997.000000      0.000000   
max       13.000000   9410.000000    4820.000000   2015.000000   2015.000000   

            zipcode           lat          long  sqft_living15     sqft_lot15  
count  21613.000000  21613.000000  21613.000000   21613.000000   21613.000000  
mean   98077.939805     47.560053   -122.213896    1986.552492   12768.455652  
std       53.505026      0.138564      0.140828     685.391304   27304.179631  
min    98001.000000     47.155900   -122.519000     399.000000     651.000000  
25%    98033.000000     47.471000   -122.328000    1490.000000    5100.000000  
50%    98065.000000     47.571800   -122.230000    1840.000000    7620.000000  
75%    98118.000000     47.678000   -122.125000    2360.000000   10083.000000  
max    98199.000000     47.777600   -121.315000    6210.000000  871200.000000

In [17]:
house.dtypes

id                 int64
date              object
price            float64
bedrooms           int64
bathrooms        float64
sqft_living        int64
sqft_lot           int64
floors           float64
waterfront         int64
view               int64
condition          int64
grade              int64
sqft_above         int64
sqft_basement      int64
yr_built           int64
yr_renovated       int64
zipcode            int64
lat              float64
long             float64
sqft_living15      int64
sqft_lot15         int64
dtype: object

In [18]:
house.isnull().count()

id               21613
date             21613
price            21613
bedrooms         21613
bathrooms        21613
sqft_living      21613
sqft_lot         21613
floors           21613
waterfront       21613
view             21613
condition        21613
grade            21613
sqft_above       21613
sqft_basement    21613
yr_built         21613
yr_renovated     21613
zipcode          21613
lat              21613
long             21613
sqft_living15    21613
sqft_lot15       21613
dtype: int64

In [55]:
house.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             21613 non-null  int64  
 1   date           21613 non-null  object 
 2   price          21613 non-null  float64
 3   bedrooms       21613 non-null  int64  
 4   bathrooms      21613 non-null  float64
 5   sqft_living    21613 non-null  int64  
 6   sqft_lot       21613 non-null  int64  
 7   floors         21613 non-null  float64
 8   waterfront     21613 non-null  int64  
 9   view           21613 non-null  int64  
 10  condition      21613 non-null  int64  
 11  grade          21613 non-null  int64  
 12  sqft_above     21613 non-null  int64  
 13  sqft_basement  21613 non-null  int64  
 14  yr_built       21613 non-null  int64  
 15  yr_renovated   21613 non-null  int64  
 16  zipcode        21613 non-null  int64  
 17  lat            21613 non-null  float64
 18  long  

Вывод: при первичном анализе данных выявлено, что датасет состоит в основном из численных данных типа int и float, 
за исключением парамерта даты продажи дома (date) - тип object. Количество строк в датасете - 21613, 
пропущенных ячеек нет.

## Задача 3.

### 3.1 В каком диапазоне изменяются стоимости недвижимости?

In [21]:
house['price']

0        221900.0
1        538000.0
2        180000.0
3        604000.0
4        510000.0
           ...   
21608    360000.0
21609    400000.0
21610    402101.0
21611    400000.0
21612    325000.0
Name: price, Length: 21613, dtype: float64

In [22]:
house['price'].min()   # минимальная цена на дом

75000.0

In [23]:
house['price'].max()   # максимальная цена на дом

7700000.0

In [19]:
house['price'].describe()

count    2.161300e+04
mean     5.400881e+05
std      3.671272e+05
min      7.500000e+04
25%      3.219500e+05
50%      4.500000e+05
75%      6.450000e+05
max      7.700000e+06
Name: price, dtype: float64

Вывод: Минимальная цена на дом составила 75 000, максимальная 7 700 000. Средняя цена датасета - 540 000.   

### 3.2 Какую долю в среднем занимают жилая площадь от всей площади по всем домам?

In [24]:
house['sqft_living']

0        1180
1        2570
2         770
3        1960
4        1680
         ... 
21608    1530
21609    2310
21610    1020
21611    1600
21612    1020
Name: sqft_living, Length: 21613, dtype: int64

In [25]:
house['sqft_lot']

0         5650
1         7242
2        10000
3         5000
4         8080
         ...  
21608     1131
21609     5813
21610     1350
21611     2388
21612     1076
Name: sqft_lot, Length: 21613, dtype: int64

In [26]:
house['sqft_living'] / house['sqft_lot'] * 100 # доля жилой квадратуры в доме

0         20.884956
1         35.487434
2          7.700000
3         39.200000
4         20.792079
            ...    
21608    135.278515
21609     39.738517
21610     75.555556
21611     67.001675
21612     94.795539
Length: 21613, dtype: float64

In [27]:
(house['sqft_living'] / house['sqft_lot'] * 100 ).mean() # средняя доля по датасету

32.37447254204737

### 3.3 Как много домов с разными этажами в данных?

In [28]:
house['floors']

0        1.0
1        2.0
2        1.0
3        1.0
4        1.0
        ... 
21608    3.0
21609    2.0
21610    2.0
21611    2.0
21612    2.0
Name: floors, Length: 21613, dtype: float64

In [31]:
house[house['floors'] >= 2]

id             date     price  bedrooms  bathrooms  \
1      6414100192  20141209T000000  538000.0         3       2.25   
6      1321400060  20140627T000000  257500.0         3       2.25   
9      3793500160  20150312T000000  323000.0         3       2.50   
15     9297300055  20150124T000000  650000.0         4       3.00   
16     1875500060  20140731T000000  395000.0         3       2.00   
...           ...              ...       ...       ...        ...   
21608   263000018  20140521T000000  360000.0         3       2.50   
21609  6600060120  20150223T000000  400000.0         4       2.50   
21610  1523300141  20140623T000000  402101.0         2       0.75   
21611   291310100  20150116T000000  400000.0         3       2.50   
21612  1523300157  20141015T000000  325000.0         2       0.75   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
1             2570      7242     2.0           0     0  ...      7   
6             1715      6819     2.0           0     0  ...      7   
9             1890      6560     2.0           0     0  ...      7   
15            2950      5000     2.0           0     3  ...      9   
16            1890     14040     2.0           0     0  ...      7   
...            ...       ...     ...         ...   ...  ...    ...   
21608         1530      1131     3.0           0     0  ...      8   
21609         2310      5813     2.0           0     0  ...      8   
21610         1020      1350     2.0           0     0  ...      7   
21611         1600      2388     2.0           0     0  ...      8   
21612         1020      1076     2.0           0     0  ...      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
1            2170            400      1951          1991    98125  47.7210   
6            1715              0      1995             0    98003  47.3097   
9            1890              0      2003             0    98038  47.3684   
15           1980            970      1979             0    98126  47.5714   
16           1890              0      1994             0    98019  47.7277   
...           ...            ...       ...           ...      ...      ...   
21608        1530              0      2009             0    98103  47.6993   
21609        2310              0      2014             0    98146  47.5107   
21610        1020              0      2009             0    98144  47.5944   
21611        1600              0      2004             0    98027  47.5345   
21612        1020              0      2008             0    98144  47.5941   

          long  sqft_living15  sqft_lot15  
1     -122.319           1690        7639  
6     -122.327           2238        6819  
9     -122.031           2390        7570  
15    -122.375           2140        4000  
16    -121.962           1890       14018  
...        ...            ...         ...  
21608 -122.346           1530        1509  
21609 -122.362           1830        7200  
21610 -122.299           1020        2007  
21611 -122.069           1410        1287  
21612 -122.299           1020        1357  

[9023 rows x 21 columns]

In [3]:
house['floors'].value_counts()

1.0    10680
2.0     8241
1.5     1910
3.0      613
2.5      161
3.5        8
Name: floors, dtype: int64

### 3.4 Насколько хорошие состояния у домов в данных?

In [33]:
house['condition'].describe()

count    21613.000000
mean         3.409430
std          0.650743
min          1.000000
25%          3.000000
50%          3.000000
75%          4.000000
max          5.000000
Name: condition, dtype: float64

In [4]:
house['condition'].value_counts()

3    14031
4     5679
5     1701
2      172
1       30
Name: condition, dtype: int64

In [38]:
house[house['condition'] <= 3]          

id             date      price  bedrooms  bathrooms  \
0      7129300520  20141013T000000   221900.0         3       1.00   
1      6414100192  20141209T000000   538000.0         3       2.25   
2      5631500400  20150225T000000   180000.0         2       1.00   
4      1954400510  20150218T000000   510000.0         3       2.00   
5      7237550310  20140512T000000  1225000.0         4       4.50   
...           ...              ...        ...       ...        ...   
21608   263000018  20140521T000000   360000.0         3       2.50   
21609  6600060120  20150223T000000   400000.0         4       2.50   
21610  1523300141  20140623T000000   402101.0         2       0.75   
21611   291310100  20150116T000000   400000.0         3       2.50   
21612  1523300157  20141015T000000   325000.0         2       0.75   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
0             1180      5650     1.0           0     0  ...      7   
1             2570      7242     2.0           0     0  ...      7   
2              770     10000     1.0           0     0  ...      6   
4             1680      8080     1.0           0     0  ...      8   
5             5420    101930     1.0           0     0  ...     11   
...            ...       ...     ...         ...   ...  ...    ...   
21608         1530      1131     3.0           0     0  ...      8   
21609         2310      5813     2.0           0     0  ...      8   
21610         1020      1350     2.0           0     0  ...      7   
21611         1600      2388     2.0           0     0  ...      8   
21612         1020      1076     2.0           0     0  ...      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
0            1180              0      1955             0    98178  47.5112   
1            2170            400      1951          1991    98125  47.7210   
2             770              0      1933             0    98028  47.7379   
4            1680              0      1987             0    98074  47.6168   
5            3890           1530      2001             0    98053  47.6561   
...           ...            ...       ...           ...      ...      ...   
21608        1530              0      2009             0    98103  47.6993   
21609        2310              0      2014             0    98146  47.5107   
21610        1020              0      2009             0    98144  47.5944   
21611        1600              0      2004             0    98027  47.5345   
21612        1020              0      2008             0    98144  47.5941   

          long  sqft_living15  sqft_lot15  
0     -122.257           1340        5650  
1     -122.319           1690        7639  
2     -122.233           2720        8062  
4     -122.045           1800        7503  
5     -122.005           4760      101930  
...        ...            ...         ...  
21608 -122.346           1530        1509  
21609 -122.362           1830        7200  
21610 -122.299           1020        2007  
21611 -122.069           1410        1287  
21612 -122.299           1020        1357  

[14233 rows x 21 columns]

In [39]:
house[house['condition'] > 3]  

id             date     price  bedrooms  bathrooms  \
3      2487200875  20141209T000000  604000.0         4       3.00   
11     9212900260  20140527T000000  468000.0         2       1.00   
12      114101516  20140528T000000  310000.0         3       1.00   
13     6054650070  20141007T000000  400000.0         3       1.75   
17     6865200140  20140529T000000  485000.0         4       1.00   
...           ...              ...       ...       ...        ...   
21079  3449000010  20150312T000000  294570.0         3       1.00   
21212  7234600098  20140905T000000  552100.0         3       3.00   
21303   522059130  20150429T000000  465000.0         3       1.00   
21332  9266700190  20150511T000000  245000.0         1       1.00   
21583  2025049203  20140610T000000  399950.0         2       1.00   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
3             1960      5000     1.0           0     0  ...      7   
11            1160      6000     1.0           0     0  ...      7   
12            1430     19901     1.5           0     0  ...      7   
13            1370      9680     1.0           0     0  ...      7   
17            1600      4300     1.5           0     0  ...      7   
...            ...       ...     ...         ...   ...  ...    ...   
21079         1140      8400     1.0           0     0  ...      7   
21212         1330      1379     2.0           0     0  ...      8   
21303         1150     18200     1.0           0     0  ...      7   
21332          390      2000     1.0           0     0  ...      6   
21583          710      1157     2.0           0     0  ...      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
3            1050            910      1965             0    98136  47.5208   
11            860            300      1942             0    98115  47.6900   
12           1430              0      1927             0    98028  47.7558   
13           1370              0      1977             0    98074  47.6127   
17           1600              0      1916             0    98103  47.6648   
...           ...            ...       ...           ...      ...      ...   
21079        1140              0      1960             0    98059  47.5022   
21212        1120            210      2005             0    98122  47.6126   
21303        1150              0      1959             0    98058  47.4262   
21332         390              0      1920             0    98103  47.6938   
21583         710              0      1943             0    98102  47.6413   

          long  sqft_living15  sqft_lot15  
3     -122.393           1360        5000  
11    -122.292           1330        6000  
12    -122.229           1780       12697  
13    -122.045           1370       10208  
17    -122.343           1610        4300  
...        ...            ...         ...  
21079 -122.144           1400        9000  
21212 -122.313           1810        1770  
21303 -122.187           1714       18200  
21332 -122.347           1340        5100  
21583 -122.329           1370        1173  

[7380 rows x 21 columns]

In [42]:
house['condition'].shape[0]

21613

In [45]:
house_good = house[house['condition'] > 3]
house_good.shape[0]

7380

In [47]:
house_good.shape[0] / house['condition'].shape[0] * 100

34.14611576366076

In [49]:
house_perfect = house[house['condition'] > 4]
house_perfect.shape[0]

1701

In [50]:
house_perfect.shape[0] / house['condition'].shape[0] * 100

7.870263267477908

In [ ]:
Вывод: Большая часть домов находится не в хорошем состоянии (по оценкам ниже либо равно 3).
Процент домов в хорошем состоянии (оценка 4 и 5) равен около 34,14%, в идеальном состоянии всего лишь 7,8%    

### 3.5 Найдите года, когда построили первый дом, когда построили последний дом в данных?

In [52]:
house['yr_built']

0        1955
1        1951
2        1933
3        1965
4        1987
         ... 
21608    2009
21609    2014
21610    2009
21611    2004
21612    2008
Name: yr_built, Length: 21613, dtype: int64

In [53]:
house['yr_built'].min() # год постройки первого дома

1900

In [54]:
house['yr_built'].max() # год постройки последнего дома

2015

## Задача 4.

### 4.1 Сколько в среднем стоят дома, у которых 2 спальни?

In [70]:
house_2bed = house[house['bedrooms'] == 2]
house_2bed['price'].mean()

401372.681884058

### 4.2 Какая в среднем общая площадь домов, у которых стоимость больше 600 000?

In [72]:
house_600K = house[house['price'] > 600000]
house_600K['sqft_lot'].mean()

20442.524776214832

### 4.3 Как много домов коснулся ремонт?

In [74]:
house.columns

Index(['id', 'date', 'price', 'bedrooms', 'bathrooms', 'sqft_living',
       'sqft_lot', 'floors', 'waterfront', 'view', 'condition', 'grade',
       'sqft_above', 'sqft_basement', 'yr_built', 'yr_renovated', 'zipcode',
       'lat', 'long', 'sqft_living15', 'sqft_lot15'],
      dtype='object')

In [75]:
house['yr_renovated']

0           0
1        1991
2           0
3           0
4           0
         ... 
21608       0
21609       0
21610       0
21611       0
21612       0
Name: yr_renovated, Length: 21613, dtype: int64

In [77]:
house[house['yr_renovated'] > 0].shape[0]

914

### 4.4 Насколько в среднем стоимость домов с оценкой grade домов выше 10 отличается от стоимости домов с оценкой grade меньше 4?

In [88]:
grade10 = house[house['grade'] > 10]
grade10['price'].mean()

1678635.1175298805

In [89]:
grade4 = house[house['grade'] < 4]
grade4['price'].mean()

189750.0

In [90]:
grade10['price'].mean() - grade4['price'].mean()

1488885.1175298805

## Задача 5.

### 5.1 Выберите дом клиенту
Клиент хочет дом с видом на набережную, как минимум с тремя ванными и с подвалом. Сколько вариантов есть у клиента?

In [97]:
# если запрос "с видом на набережную" соответствует оценке вида 4 и при этом наличие набережной
client_house = house[(house['waterfront'] == True) & (house['view'] == 4) & (house['bathrooms'] >= 3) & (house['sqft_basement'] > 0)]
client_house.shape[0]

36

### 5.2 Выберите дом клиенту
Клиент хочет дом либо с очень красивым видом из окна, либо с видом на набережную, в очень хорошем состоянии и год постройки не меньше 1980 года. В какой ценовом диапазоне будут дома?

In [101]:
house1980_good = house[(house['yr_built'] >= 1980) & (house['condition'] > 4)]
client_house = house1980_good[(house1980_good['view'] == 4) | (house1980_good['waterfront'] == True)]
client_house['price'].min(), client_house['price'].max()

(1295000.0, 3000000.0)

### 5.3 Выберите дом клиенту
Клиент хочет дом без подвала, с двумя этажами, стоимостью до 150000. Какая оценка по состоянию у таких домов в среднем?

In [102]:
client_house = house[(house['floors'] == 2) & (house['sqft_basement'] == 0) & (house['price'] < 150000)]
client_house

id             date     price  bedrooms  bathrooms  \
2589   5061300030  20140508T000000  134000.0         2       1.50   
10105  5466310060  20150324T000000  139500.0         2       1.50   
15610  7614100080  20150211T000000  140000.0         3       1.75   
18332  9272202260  20140924T000000  130000.0         3       1.00   
18939  4239400300  20141129T000000   90000.0         3       1.00   
18988  7399100210  20141126T000000  140000.0         3       1.50   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
2589           980      5000     2.0           0     0  ...      7   
10105         1230      1561     2.0           0     0  ...      7   
15610         1270      8991     2.0           0     0  ...      7   
18332         1200      7000     2.0           0     0  ...      7   
18939          980      2490     2.0           0     0  ...      6   
18988         1200      2002     2.0           0     0  ...      8   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
2589          980              0      1922          2003    98014  47.7076   
10105        1230              0      1983             0    98042  47.3768   
15610        1270              0      1981             0    98042  47.3563   
18332        1200              0      1908             0    98116  47.5883   
18939         980              0      1969             0    98092  47.3170   
18988        1200              0      1966             0    98055  47.4659   

          long  sqft_living15  sqft_lot15  
2589  -121.359           1040        5000  
10105 -122.149           1660        2243  
15610 -122.149           1270        8993  
18332 -122.384           3290        6000  
18939 -122.182            980        3154  
18988 -122.189           1270        1848  

[6 rows x 21 columns]

In [103]:
client_house['condition'].mean()

2.8333333333333335